### PR TITLE
Xml

### PR DESCRIPTION
**Pull request by nickus**
#### What's this PR do/fix?
This PR adds XML supporting for models. 
https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#xmlObject
#### What are the relevant issues?

### CODE REVIEW

**Comment by dilipkrish** [comment]
@nickus thanks for the PR. Not to get stuck on coverage numbers alone, but it would be useful to add some tests.

**Comment by dilipkrish** [comment]
This is related to #1902 and #1903

**Comment by nickus** [comment]
OK

**Comment by nickus** [comment]
done

File: springfox-schema/src/main/java/springfox/documentation/schema/DefaultModelProvider.java [hunk 1]
```diff
@@ -109,10 +109,20 @@ private Model modelBuilder(ResolvedType propertiesHost,
...
     return schemaPluginsManager.model(modelContext);
   }
 
+  private Xml buildXml() {
+    return new Xml()
+        .name("")
+        .attribute(false)
+        .namespace("")
+        .prefix("")
+        .wrapped(false);
```
Comment (dilipkrish) [inline springfox-schema/src/main/java/springfox/documentation/schema/DefaultModelProvider.java]: This doesnt seem right. We shouldn't always initialize the xml. this has resulted in all the contracts having the xml element in them.

Also we need a way to grab the right xml namespace/prefix/name from the `XmlSchema` annotation.

File: springfox-core/src/main/java/springfox/documentation/builders/ModelPropertyBuilder.java [hunk 2]
```diff
@@ -103,6 +105,12 @@ public ModelPropertyBuilder extensions(List<VendorExtension> extensions) {
     return this;
   }
 
+  public ModelPropertyBuilder xml(Xml xml) {
+    this.xml = defaultIfAbsent(xml, this.xml);
+    return this;
+  }
```
Comment (dilipkrish) [inline springfox-core/src/main/java/springfox/documentation/builders/ModelPropertyBuilder.java]: This method is ever used. Was that intentional?

**Comment by dilipkrish** [comment]
@nickus I polished and merged your PR. Yet to be done for complete support is reading the XmlSchema by package and adding support fo xmlns prefixes. Would love it if you could complete the work for Xml support.

**Comment by dilipkrish** [comment]
I created an XmlModelPlugin

https://github.com/springfox/springfox/blob/640c75a6900fa284c754e7dc4cc4c3982e7f4469/springfox-schema/src/main/java/springfox/documentation/schema/XmlModelPlugin.java#L42-L45

and an XmlPropertyPlugin

https://github.com/springfox/springfox/blob/master/springfox-schema/src/main/java/springfox/documentation/schema/property/XmlPropertyPlugin.java#L43-L82

Would be awesome if you verified the desired output as well

**Comment by jgaribay21** [comment]
looks like there is a missing check optional check here:
`  Optional<XmlElementWrapper> wrapper = findPropertyAnnotation( 
       context.getBeanPropertyDefinition().get(), 
       XmlElementWrapper.class); `

If it is a field that has the XmlElement annotation this will cause an IllegalStateException.  It should be checking if it is a bean, and if not if the wrapper is annotated on the field:

`@XmlElementWrapper(name = "TestElements")
@XmlElement(name = "TestElement")
protected List<SomeObject> testElements = new ArrayList<SomeObject>();`

**Comment by jgaribay21** [comment]
Also, XML properties are lost when running through ApiModelReader.mergeModelMap.  The merge should copy over the Xml node to the new modelbuilder.

**Comment by dilipkrish** [comment]
@jgaribay21 thanks for the feedback! this is helpful. Would you mind creating an issue for it?

**Pull request merged**